internal/os/exec/lp_windows.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package exec
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func chkStat(file string) error {
+	d, err := os.Stat(file)
+	if err != nil {
+		return err
+	}
+	if d.IsDir() {
+		return os.ErrPermission
+	}
+	return nil
+}
+
+func hasExt(file string) bool {
+	i := strings.LastIndex(file, ".")
+	if i < 0 {
+		return false
+	}
+	return strings.LastIndexAny(file, `:\/`) < i
+}
+
+func findExecutable(file string, exts []string) (string, error) {
+	if len(exts) == 0 {
+		return file, chkStat(file)
+	}
+	if hasExt(file) {
+		if chkStat(file) == nil {
+			return file, nil
+		}
+	}
+	for _, e := range exts {
+		if f := file + e; chkStat(f) == nil {
+			return f, nil
+		}
+	}
+	return "", os.ErrNotExist
+}
+
+// LookPath searches for an executable named file in the
+// directories named by the PATH environment variable.
+// If file contains a slash, it is tried directly and the PATH is not consulted.
+// LookPath also uses PATHEXT environment variable to match
+// a suitable candidate.
+// The result may be an absolute path or a path relative to the current directory.
+func LookPath(path, file string) (string, error) {
```

Comment:
Rather than passing in path directly, and given that different systems do different things (e.g. TEXTEXT), how about passing in a getenv function?

    // LookPath searches for an executable named file, using getenv
    // to look up environment variables. If getenv is nil, os.Getenv
    // will be used. If file contains a slash, it is tried directly and
    // getenv will not be called.
    // The result may be an absolute path or a path relative to the current directory.
    func LookPath(file string, getenv func(string) string) (string, error)

?

Reply:
Done. I've also used an existing method on `TestScript` as part of this, so turned out to be a nice change.